pkg/txn/storage/txn/batch.go

```diff
@@ -28,8 +28,15 @@ type BatchIter func() (tuple []Nullable)
 func NewBatchIter(b *batch.Batch) BatchIter {
 	i := 0
 	iter := func() (tuple []Nullable) {
-		if i >= b.Vecs[0].Length() {
-			return
+		for {
```

Comment:
I am not a fan of the Iter idiom.    Caller/Caller inversed and there is no way to vectorwise anything.

Reply:
The mem engine is row-based, so batches need to be iterated by rows to operate.